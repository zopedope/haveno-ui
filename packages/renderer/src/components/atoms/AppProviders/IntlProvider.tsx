// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { useEffect, useMemo, useState } from "react";
import { IntlProvider as ReacIntlProvider } from "react-intl";
import type { FC } from "react";
import { LangPack, SupportedLocales } from "@src/constants/lang";
import type { LangKeys } from "@src/constants/lang";

const DEFAULT_LOCALE = SupportedLocales.EN;

export const IntlProvider: FC = ({ children }) => {
  const [messages, setMessages] = useState<{ [key in LangKeys]: string }>();
  const locale = useMemo(
    () =>
      navigator.language in SupportedLocales
        ? navigator.language
        : DEFAULT_LOCALE,
    [navigator.language]
  );

  useEffect(() => {
    setMessages(LangPack[locale] ?? LangPack[DEFAULT_LOCALE]);
  }, [locale]);

  return (
    <ReacIntlProvider locale={locale} messages={messages}>
      {children}
    </ReacIntlProvider>
  );
};
