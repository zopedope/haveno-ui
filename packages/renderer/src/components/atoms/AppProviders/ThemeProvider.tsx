// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { MantineProvider, Global } from "@mantine/core";
import type { FC } from "react";
import { themeOverride, globalStyles } from "@src/theme";

export const ThemeProvider: FC = ({ children }) => {
  return (
    <MantineProvider theme={themeOverride}>
      <Global styles={globalStyles} />
      {children}
    </MantineProvider>
  );
};
