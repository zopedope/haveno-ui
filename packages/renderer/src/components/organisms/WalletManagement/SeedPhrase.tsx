// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import {
  createStyles,
  Group,
  List,
  ListItem,
  Space,
  Stack,
} from "@mantine/core";
import { BodyText } from "@atoms/Typography";
import { useXmrSeed } from "@hooks/haveno/useXmrSeed";

export function SeedPhrase() {
  const { classes } = useStyles();
  const { data: xmrseeds, isLoading } = useXmrSeed();
  const seeds = xmrseeds?.split(" ");

  return (
    <Stack>
      <Space h="lg" />
      {isLoading && <BodyText>Loading Wallet Seeds...</BodyText>}
      <List type="ordered" className={classes.container}>
        {seeds?.map((label, index) => (
          <Group key={index} className={classes.background} spacing="sm">
            <ListItem>{label}</ListItem>
          </Group>
        ))}
      </List>
    </Stack>
  );
}

const useStyles = createStyles((theme) => ({
  background: {
    backgroundColor: theme.colors.gray[3],
    borderRadius: "0.5rem",
    flex: "0 30%",
    li: {
      color: theme.colors.gray[6],
    },
    padding: "0.75rem 1rem",
    span: {
      color: theme.colors.gray[9],
    },
    width: "15rem",
  },
  container: {
    display: "flex",
    flexWrap: "wrap",
    gap: "1rem",
  },
}));
