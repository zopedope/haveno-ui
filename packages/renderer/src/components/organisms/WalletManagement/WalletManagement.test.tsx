// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { beforeAll, describe, expect, it, vi } from "vitest";
import { render, waitForElementToBeRemoved } from "@testing-library/react";
import { SeedPhrase } from "./SeedPhrase";
import { WalletManagement } from ".";
import { AppProviders } from "@atoms/AppProviders";

describe("molecules::WalletManagement", () => {
  beforeAll(() => {
    vi.mock("@hooks/haveno/useValidatePassword", () => ({
      useValidatePassword: () => ({
        isLoading: false,
        isSuccess: true,
        data: true,
      }),
    }));

    vi.mock("@hooks/haveno/useXmrSeed", () => ({
      useXmrSeed: () => ({
        isLoading: true,
        isSuccess: true,
        data: "baptism ounce solved gimmick cafe absorb pouch gesture fawns degrees bikini inline island oncoming menu tissue cajun inwardly chlorine popular sleepless taboo aces arises popular",
      }),
    }));
  });

  it("renders loading state", () => {
    const { asFragment } = render(
      <AppProviders>
        <WalletManagement />
      </AppProviders>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders after loading data", async () => {
    const { asFragment, queryByText } = render(
      <AppProviders>
        <SeedPhrase />
      </AppProviders>
    );
    if (queryByText("Loading...")) {
      await waitForElementToBeRemoved(() => queryByText("Loading..."));
    }
    expect(asFragment()).toMatchSnapshot();
  });
});
