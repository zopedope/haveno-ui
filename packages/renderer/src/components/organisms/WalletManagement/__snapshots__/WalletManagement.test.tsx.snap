// Vitest Snapshot v1

exports[`molecules::WalletManagement > renders after loading data 1`] = `
<DocumentFragment>
  <div
    class="mantine-Stack-root mantine-lfk3cq"
  >
    <div
      class="mantine-63n06h"
    />
    <div
      class="mantine-Text-root mantine-1r7efq5"
    >
      Loading Wallet Seeds...
    </div>
    <ol
      class="mantine-List-root mantine-shpils"
    >
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              baptism
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              ounce
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              solved
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              gimmick
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              cafe
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              absorb
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              pouch
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              gesture
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              fawns
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              degrees
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              bikini
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              inline
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              island
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              oncoming
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              menu
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              tissue
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              cajun
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              inwardly
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              chlorine
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              popular
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              sleepless
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              taboo
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              aces
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              arises
            </span>
          </div>
        </li>
      </div>
      <div
        class="mantine-Group-root mantine-1qkd5ve"
      >
        <li
          class="mantine-List-item mantine-Group-child mantine-xm6zd"
        >
          <div
            class="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
          >
            <span>
              popular
            </span>
          </div>
        </li>
      </div>
    </ol>
  </div>
</DocumentFragment>
`;

exports[`molecules::WalletManagement > renders loading state 1`] = `
<DocumentFragment>
  <div
    class="mantine-Stack-root mantine-lfk3cq"
  >
    <form>
      <div
        class="mantine-Stack-root mantine-lfk3cq"
      >
        <div
          class="mantine-78ek8g"
        />
        <div
          class="mantine-PasswordInput-root mantine-1u48g6v"
        >
          <label
            class="mantine-PasswordInput-label mantine-1bjo575"
            for="password"
            id="password-label"
          >
            Password
            <span
              aria-hidden="true"
              class="mantine-1wc35tu mantine-PasswordInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-PasswordInput-wrapper mantine-12sbrde"
          >
            <div
              aria-invalid="false"
              class="mantine-PasswordInput-defaultVariant mantine-PasswordInput-input mantine-PasswordInput-input mantine-1hlli39"
            >
              <input
                class="mantine-PasswordInput-innerInput mantine-66fxec"
                id="password"
                required=""
                type="password"
                value=""
              />
            </div>
            <div
              class="mantine-o3oqoy mantine-PasswordInput-rightSection"
            >
              <button
                aria-hidden="true"
                class="mantine-ActionIcon-hover mantine-ActionIcon-root mantine-PasswordInput-visibilityToggle mantine-11pctd2"
                tabindex="-1"
                type="button"
              >
                <svg
                  fill="none"
                  height="15"
                  viewBox="0 0 15 15"
                  width="15"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M7.5 11C4.80285 11 2.52952 9.62184 1.09622 7.50001C2.52952 5.37816 4.80285 4 7.5 4C10.1971 4 12.4705 5.37816 13.9038 7.50001C12.4705 9.62183 10.1971 11 7.5 11ZM7.5 3C4.30786 3 1.65639 4.70638 0.0760002 7.23501C-0.0253338 7.39715 -0.0253334 7.60288 0.0760014 7.76501C1.65639 10.2936 4.30786 12 7.5 12C10.6921 12 13.3436 10.2936 14.924 7.76501C15.0253 7.60288 15.0253 7.39715 14.924 7.23501C13.3436 4.70638 10.6921 3 7.5 3ZM7.5 9.5C8.60457 9.5 9.5 8.60457 9.5 7.5C9.5 6.39543 8.60457 5.5 7.5 5.5C6.39543 5.5 5.5 6.39543 5.5 7.5C5.5 8.60457 6.39543 9.5 7.5 9.5Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="mantine-63n06h"
        />
        <div
          class="mantine-Group-root mantine-147cgkf"
        >
          <button
            class="mantine-Button-filled mantine-Button-root mantine-Group-child mantine-18llzhv"
            disabled=""
            type="submit"
          >
            <div
              class="mantine-3xbgk5 mantine-Button-inner"
            >
              <span
                class="mantine-qo1k2 mantine-Button-label"
              >
                Reveal Seed Phrase
              </span>
            </div>
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
